### PR TITLE
ORSP-3 Create a mechanism for ORSP Admins to manually assign an Admin's name (Andrea, Susie, Emily, Kelly) 

### PR DESCRIPTION
**Pull request by verovicario**
## Addresses
https://broadinstitute.atlassian.net/browse/ORSP-3
## Changes
Add a new column only available for ORSP Admins (Andrea, Susie, Emily, Kelly) in My Task List and My Project List. 
## Testing
Go to My Task List and my project list from index page.  Assigned Admin column shouldn't been displayed.
ymls have been updated in google's project to include the following conf:
orspAdmins: [saltzman, sweisenb, lipscomb, kflannig]
In order to test, add your userName to be able to see the new column in the application.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from ORSP representative
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/controllers/org/broadinstitute/orsp/SearchController.groovy [hunk 1]
```diff
@@ -216,4 +216,17 @@ class SearchController implements UserInfo {
         render([text: data as JSON, contentType: "application/json"])
     }
 
+    def getORSPAdmins() {
+        def response = []
+        userService.findUsers(Arrays.asList(grailsApplication.config.getProperty('orspAdmins')?.split(",")))
+        .each {
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/SearchController.groovy]: nit: indentation

File: grails-app/controllers/org/broadinstitute/orsp/UserController.groovy [hunk 2]
```diff
@@ -72,4 +72,10 @@ class UserController extends AuthenticatedController {
             render([error: e.message] as JSON)
         }
     }
+
+
+    def isORSPAdmin() {
+        List<String> orspAdminUserNames = Arrays.asList(grailsApplication.config.getProperty('orspAdmins')?.split(","))
+        render (["isORSPAdmin": orspAdminUserNames.find {getUser().userName.equals(it)}?.length() > 0] as JSON)
+    }
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/UserController.groovy]: nit: line breaks, should be one before and one after.

File: grails-app/controllers/org/broadinstitute/orsp/UserController.groovy [hunk 3]
```diff
@@ -72,4 +72,10 @@ class UserController extends AuthenticatedController {
             render([error: e.message] as JSON)
         }
     }
+
+
+    def isORSPAdmin() {
+        List<String> orspAdminUserNames = Arrays.asList(grailsApplication.config.getProperty('orspAdmins')?.split(","))
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/UserController.groovy]: Since you're pulling this (`grailsApplication.config.getProperty('orspAdmins')`) in two places now, does it make sense to pull it up into `AuthenticatedController` so it can be loaded once and for all?

File: grails-app/controllers/org/broadinstitute/orsp/api/ProjectController.groovy [hunk 4]
```diff
@@ -69,6 +69,17 @@ class ProjectController extends AuthenticatedController {
         }
     }
 
+
+    def removeAssignedAdmin() {
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/api/ProjectController.groovy:73]: Does this need any user validation? Can any one call this endpoint with anything they want and unset the admin? This seems like a security hole.

File: grails-app/services/org/broadinstitute/orsp/IssueService.groovy [hunk 5]
```diff
@@ -363,6 +364,17 @@ class IssueService implements UserInfo {
         issue
     }
 
+    @SuppressWarnings(["GroovyAssignabilityCheck"])
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/IssueService.groovy]: Doesn't look like you need this warning suppression. Copy/paste?

File: grails-app/services/org/broadinstitute/orsp/IssueService.groovy [hunk 6]
```diff
@@ -363,6 +364,17 @@ class IssueService implements UserInfo {
...
+        if(property != null) {
+           issue.removeFromExtraProperties(property)
+           property.delete(hard: true, flush: true)
+        }
+        issue
+    }
+
     // Todo This method doesn't handle funding changes. We should handle it when implementing edit functions
     @SuppressWarnings(["GroovyMissingReturnStatement"])
     Issue modifyIssueProperties (Issue issue, Object input) {
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/IssueService.groovy]: Not part of this PR, but this method looks unused now.

File: grails-app/controllers/org/broadinstitute/orsp/SearchController.groovy [hunk 7]
```diff
@@ -216,4 +216,17 @@ class SearchController implements UserInfo {
         render([text: data as JSON, contentType: "application/json"])
     }
 
+    def getORSPAdmins() {
+        def response = []
+        userService.findUsers(Arrays.asList(grailsApplication.config.getProperty('orspAdmins')?.split(",")))
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/SearchController.groovy]: see comment on `UserController`

File: src/main/webapp/components/UserListDialog.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+import { Modal, ModalBody, ModalFooter, ModalHeader, ModalTitle } from 'react-bootstrap';
+import { InputFieldSelect } from '../components/InputFieldSelect';
+import { Project, Search } from '../util/ajax';
+import './ConfirmationDialog.css';
+import LoadingWrapper from './LoadingWrapper';
+import isEmpty from 'lodash/isEmpty';
+
+const UserListDialog = hh(class UserListDialog extends Component {
+
+  _isMounted = false;
```
Comment (rushtong) [inline src/main/webapp/components/UserListDialog.js]: Is this used? IIRC, manually tracking mounted state is a react anti-pattern.

File: src/main/webapp/issueList/IssueList.js [hunk 9]
```diff
@@ -133,25 +184,38 @@ const IssueList = hh(class IssueList extends Component {
...
 
-  tableHandler = () => {
-      Project.getProjectByUser(this.paramsContext.get('assignee'), this.paramsContext.get('max')).then(result => {
-        if(this._isMounted) {
-          this.setState(prev => {
-            prev.issues = result.data;
-            return prev;
-          }, () => this.props.hideSpinner())
-        }   
+  tableHandler = async() => {
```
Comment (rushtong) [inline src/main/webapp/issueList/IssueList.js]: nit: here and throughout the file, the indentation and spacing is off. I suggest using the IDE's reformat code feature which will apply the eslint styles.

**Review by rushtong — COMMENTED** [review]
See comments inline. Unsure about the security aspect of removing assigned admins - please follow up with ORSP team on what the usage restrictions should be around that feature.

**Comment by verovicario** [comment]
Hi @rushtong , 
I talked this morning with ORSP team. They decided to allow admins to assign and remove ORSP admins. 
Users who aren't admins only can see the column but they can't perform any action.
I worked on your feedback. If you have any other comments, please let me know. Thanks!

File: grails-app/controllers/org/broadinstitute/orsp/UserController.groovy (hunk 2 above)
Comment (verovicario) [reply grails-app/controllers/org/broadinstitute/orsp/UserController.groovy]: I removed this method, the new column will be display for all admins, so this method is not necessary

File: grails-app/controllers/org/broadinstitute/orsp/api/ProjectController.groovy (hunk 4 above)
Comment (verovicario) [reply grails-app/controllers/org/broadinstitute/orsp/api/ProjectController.groovy:73]: I added a validation to allow only admins to perform this action

**Review by rushtong — APPROVED** [review]
